dps/utils/systematic.py

```diff
@@ -463,59 +466,130 @@ def print_dictionary(title, dictionary_to_print):
+        table_outfile = table_outfile_tmp.format( ch=channel, sys=syst_name, var=variable, label='Covariance')
+        create_covariance_matrix(covariance_matrix, table_outfile)
+        make_covariance_plot(options, syst_name, covariance_matrix)
+        table_outfile = table_outfile_tmp.format( ch=channel, sys=syst_name, var=variable, label='Correlation')
+        create_covariance_matrix(correlation_matrix, table_outfile)
+        make_covariance_plot(options, syst_name, correlation_matrix, label='Correlation')
+
+    generate_total_covariance(options, all_covariance_matrices)
 
-    for systematic, measurement in x_sec_with_symmetrised_systematics.iteritems():
-        if systematic == 'central': continue
-        covariance_matrix, correlation_matrix = generate_covariance_matrix(number_of_bins, systematic, measurement)
-        make_covariance_plot(options, systematic, covariance_matrix)
-        make_covariance_plot(options, systematic, correlation_matrix, label='Correlation')
     return
 
-def generate_covariance_matrix(number_of_bins, systematic, measurement):
+def generate_covariance_matrix(number_of_bins, systematic, sign, statistic):
     '''
-    Covariance_ij = (Sign_i*Unc_i) * (Sign_j*Unc_j)
     Variance_ii = (Unc_i) * (Unc_i)
-    Returns the matrix in the form [[Bin_i, Bin_j], Cov_ij]
+    Correlation_ij = (Sign_i) * (Sign_j)
+    Covariance_ij = (Sign_i*Unc_i) * (Sign_j*Unc_j)
+    Returns the matrix in the form of a numpy matrix    
     '''
-    covariance_matrix = []
-    correlation_matrix = []
+    cov_matrix = []
+    cor_matrix = []
     for bin_i in xrange(number_of_bins):
+        cov_matrix_row=[]
+        cor_matrix_row=[]
         for bin_j in xrange(number_of_bins):
-            if (bin_j < bin_i): continue
-            uncertainty_i = measurement[0][bin_i]
-            uncertainty_j = measurement[0][bin_j]
-            sign_i = measurement[1][bin_i]
-            sign_j = measurement[1][bin_j]
-            cov_ij = (sign_i*uncertainty_i)*(sign_j*uncertainty_j)
-            cor_ij = ( sign_i * sign_j )
-            # Bins when plotting Histogram start from 1 not 0
-            bin_and_value_covariance = [[bin_i+1, bin_j+1], cov_ij]
-            covariance_matrix.append(bin_and_value_covariance)
-            bin_and_value_correlation = [[bin_i+1, bin_j+1], cor_ij]
-            correlation_matrix.append(bin_and_value_correlation)
-            if not bin_i == bin_j:
-                bin_and_value = ([[bin_j+1, bin_i+1], cov_ij])
-                covariance_matrix.append(bin_and_value)
-                bin_and_value_correlation = [[bin_j+1, bin_i+1], cor_ij]
-                correlation_matrix.append(bin_and_value_correlation)
+            uncertainty_i   = systematic[bin_i]
+            uncertainty_j   = systematic[bin_j]
+            sign_i          = sign[bin_i]
+            sign_j          = sign[bin_j]
+            stat_i          = statistic[bin_i][1]
+            stat_j          = statistic[bin_j][1]     
```

Comment:
ah yes well spotted. It was from an earlier attempt to find the covariance/correlation from the statistical uncertainties, like in the systematic case, before I used the TUnfold. I shall remove